### PR TITLE
Rails Blog #5: Controllers Specs

### PR DESCRIPTION
**Pull request by danifromecuador**
## In this pull request, I've:
- Created a `4-controllers-specs` branch
- Installed `rspec-rails` gem
- Created specs for `users` and `posts` to tests it's `index` and `show`

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  describe 'GET /index' do
+    it 'renders an h2 text' do
+      get "/users/#{first_user.id}/posts"
+
+      expect(response.body).to include('<h2>List of all posts for a given user</h2>')
+    end
+  end
+
```
Comment (Davidosky007) [inline spec/requests/posts_spec.rb]: * Good job writing tests for each controller but as mentioned in the [requirements ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text)
```
Make sure that for each action you check:
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text
```
* Kindly write all these tests for each action for example post controller(index action has 3 tests, while show action has 3 tests too) all tests should cover the required specifications

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @danifromecuador 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Tests written for all controllers 👍 
* Well-written PR 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (danifromecuador) [reply spec/requests/posts_spec.rb]: thank you, dear @Davidosky007, I have created 3 tests for `index` and 3 tests for `show`, for both, `users` and `posts` so now there is 12 tests in total 
image

**Comment by danifromecuador** [comment]
Dear code reviewer, could you please review the models and DB?, I don't know if all the relationships and methods are right. Thank you in advance for your detailed review.

**Comment by prolajumokeoni** [comment]
Hi @danifromecuador, for this milestone you don't need to setup the models or db, in the creating data model milestone you will work on this, for now everything looks great 👍🏿

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @danifromecuador, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request. ✔️
- Controller-spec implemented ✔️
- `user_spec.rb` & `post_spec.rb` file seen ✔️ 
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**